x-pack/plugins/apm/public/context/breadcrumbs/use_breadcrumb.ts

```diff
@@ -21,22 +21,25 @@ export function useBreadcrumb(breadcrumb: Breadcrumb | Breadcrumb[]) {
 
   const matchedRoute = useRef(match?.route);
 
-  useEffect(() => {
-    if (matchedRoute.current && matchedRoute.current !== match?.route) {
-      api.unset(matchedRoute.current);
-    }
-
-    matchedRoute.current = match?.route;
+  useEffect(
```

Comment:
Put the `useEffect` back, one of the unit tests failed because of:

```
Warning: Cannot update a component (`BreadcrumbsContextProvider`) while rendering a different component (`Breadcrumb`). To locate the bad setState() call inside `Breadcrumb`, follow the stack trace as described in https://fb.me/setstate-in-render
        in Breadcrumb
        in OutletContextProvider (created by CurrentRouteContextProvider)
        in CurrentRouteContextProvider (created by RouteRenderer)
        in RouteRenderer (created by ApmAppRoot)
        in ThemeProvider (created by ApmThemeProvider)
        in ApmThemeProvider (created by ApmAppRoot)
```

Reply:
It's because without the useEffect the api.set is called immediately after the useBreadcrumb is invoked updating the BreadcrumbsContextProvider, but that happens while the Breadcrumb is still rendering, throwing this message.